packages/react-instantsearch-hooks-server/src/__tests__/modules-none.test.tsx

```diff
@@ -114,9 +114,21 @@ describe('ReactDOMServer imports', () => {
     await expect(
       getServerState(<App />)
     ).rejects.toThrowErrorMatchingInlineSnapshot(
-      `"Could not import ReactDOMServer."`
+      `"Could not import ReactDOMServer. You can provide it as an argument: getServerState(<Search />, ReactDOMServer.renderToString)."`
     );
   });
+
+  test('does not throw if user provides their own renderToString', async () => {
+    const searchClient = createSearchClient({});
+    const { App } = createTestEnvironment({ searchClient });
+
+    const serverState = await getServerState(<App />, {
+      renderToString: (element) =>
+        jest.requireActual('react-dom/server').renderToString(element),
+    });
+
+    expect(serverState.initialResults).toEqual(expect.any(Object));
+  });
```

Comment:
Here we could rather test that the provided `renderToString` is called when calling `getServerState(<App />, { renderToString })`.

Asserting that it doesn't throw doesn't add as much value.